FormalConjectures/Paper/DegreeSequencesTriangleFree.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+import FormalConjectures.ForMathlib.Combinatorics.SimpleGraph.DiamExtra
+import FormalConjectures.ForMathlib.Combinatorics.SimpleGraph.GraphConjectures.Domination
+import FormalConjectures.ForMathlib.Combinatorics.SimpleGraph.Bipartite
+import FormalConjectures.ForMathlib.Combinatorics.SimpleGraph.GraphConjectures.Definitions
+import FormalConjectures.ForMathlib.Combinatorics.SimpleGraph.GraphConjectures.Invariants
+
+/-!
+Title: Degree sequences in triangle-free graphs
+Authors: P. Erdős, S. Fajtlowicz and W. Staton,
+Published in Discrete Mathematics 92 (1991) 85–88.
+-/
+
+open BigOperators
+open Classical
+
+/-- A sequence of natural numbers is **compact** if consecutive terms at distance
+`2` differ by `1`. -/
+def IsCompactSequence (d : ℕ → ℕ) : Prop :=
+  ∀ k, d (k + 2) = d k + 1
+
+namespace SimpleGraph
+
+variable {α : Type*} [Fintype α] [DecidableEq α]
+
+/-- The number of vertices of `G` having degree `d`. -/
+noncomputable def degreeFreq (G : SimpleGraph α) (d : ℕ) : ℕ :=
+  (Finset.univ.filter fun v : α => G.degree v = d).card
+
+/-- `δ G` is the minimum degree of `G`. -/
+noncomputable def δ (G : SimpleGraph α) [DecidableRel G.Adj] : ℕ :=
+  sInf (Set.range fun v : α => G.degree v)
+
+end SimpleGraph
+
+section lemmas
+
+variable (d : ℕ → ℕ) (n k r : ℕ)
```

Comment:
Here the paper also assumes that `d` is nondecreasing, that all terms are positive and that no three terms are equal.